### PR TITLE
Faster is_blank_line() implementation

### PR DESCRIPTION
**Pull request by Explorer09**
Introduce static is_blank_line() function in filter.c.
Remove regex_blank_line and use is_blank_line() in place of its role.

### CODE REVIEW

File: src/filter.c [hunk 1]
```diff
@@ -326,7 +326,15 @@ int filter_tee_header (struct filter *chain)
 
 static bool is_blank_line (const char *str)
 {
-	return (regexec (&regex_blank_line, str, 0, NULL, 0) == 0);
+#if 1 /* faster */
+	while (isspace(*str))
+		str++;
+	return (*str == '\0');
+#endif
+#if 0 /* smaller */
```
Comment (westes) [inline src/filter.c]: What is the purpose of the #if 1 / #if 0?

Comment (Explorer09) [inline src/filter.c]: These are just two implementations that you can choose one. Both are faster and take smaller code than the current regex approach.
If you think there is no need to keep two pieces of (functionally equivalent) code, feel free to delete one of them, and have the other into master.

**Comment by westes** [comment]
I will pick one of them and declutter the #if 1/0.

On Friday, 14 July 2017,  7:30 pm -0700, "Kang-Che Sung (宋岡哲)" <notifications@github.com> wrote:

> Explorer09 commented on this pull request.
> 
> 
> 
> > @@ -326,7 +326,15 @@ int filter_tee_header (struct filter *chain)
>  
>  static bool is_blank_line (const char *str)
>  {
> -	return (regexec (&regex_blank_line, str, 0, NULL, 0) == 0);
> +#if 1 /* faster */
> +	while (isspace(*str))
> +		str++;
> +	return (*str == '\0');
> +#endif
> +#if 0 /* smaller */
> 
> These are just two implementations that you can choose one. Both are faster and take smaller code than the current regex approach.
> If you think there is no need to keep two pieces of (functionally equivalent) code, feel free to delete one of them, and have the other into master.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/westes/flex/pull/237#discussion_r127574457

-- 
Will Estes
westes575@gmail.com

**Comment by westes** [comment]
Thanks. This is on master -- with some cleanup.

**Pull request closed without merging**